instance_clusters.go

```diff
@@ -0,0 +1,206 @@
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"time"
+)
+
+// InstanceClustersService handles communication with the
+// instance clusters related methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceClustersService struct {
+	client *Client
+}
+
+// InstanceCluster represents a GitLab Instance Cluster.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/instance_clusters.html
+type InstanceCluster struct {
+	ID                 int                 `json:"id"`
+	Name               string              `json:"name"`
+	Domain             string              `json:"domain"`
+	CreatedAt          *time.Time          `json:"created_at"`
+	ProviderType       string              `json:"provider_type"`
+	PlatformType       string              `json:"platform_type"`
+	EnvironmentScope   string              `json:"environment_scope"`
+	ClusterType        string              `json:"cluster_type"`
+	User               *User               `json:"user"`
+	PlatformKubernetes *PlatformKubernetes `json:"platform_kubernetes"`
+	ManagementProject  *ManagementProject  `json:"management_project"`
+}
+
+func (v InstanceCluster) String() string {
+	return Stringify(v)
+}
+
+// PlatformKubernetes represents a GitLab Instance Cluster PlatformKubernetes.
+type InstanceClusterPlatformKubernetes struct {
+	APIURL            string `json:"api_url"`
+	Token             string `json:"token"`
+	CaCert            string `json:"ca_cert"`
+	Namespace         string `json:"namespace"`
+	AuthorizationType string `json:"authorization_type"`
+}
+
+// ListClusters gets a list of all instance clusters.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#list-instance-clusters
+func (s *InstanceClustersService) ListClusters(options ...RequestOptionFunc) ([]*InstanceCluster, *Response, error) {
+	u := fmt.Sprintf("admin/clusters")
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ics []*InstanceCluster
+	resp, err := s.client.Do(req, &ics)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ics, resp, err
+}
+
+// GetCluster gets an instance cluster.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#get-a-single-instance-cluster
+func (s *InstanceClustersService) GetCluster(clusterID int, options ...RequestOptionFunc) (*InstanceCluster, *Response, error) {
+	u := fmt.Sprintf("admin/clusters/%d", clusterID)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ic := new(InstanceCluster)
+	resp, err := s.client.Do(req, &ic)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ic, resp, err
+}
+
+// AddInstanceClusterOptions represents the available AddCluster() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/instance_clusters.html#add-existing-cluster-to-instance
+type AddInstanceClusterOptions struct {
```

Comment:
```suggestion
type AddClusterOptions struct {
```

Reply:
`AddClusterOptions` is already declared in `https://github.com/xanzy/go-gitlab/blob/d3946d4f8c62bfae945062be289f13ea5e6ecfe4/project_clusters.go#L129`

Reply:
Same for this one, if the struct already exists, why not reuse it here versus declaring the same struct twice?